### PR TITLE
WFCORE-5493: Invoke DeploymentTransformer in operation "full-replace-…

### PR DESCRIPTION
**Pull request by fanste**
…deployment"

https://issues.redhat.com/browse/WFCORE-5493

### CODE REVIEW

**Review by ehsavoie — CHANGES_REQUESTED** [review]
I think the code should be shared instead of being copied from DeploymentAddHandler

**Comment by ehsavoie** [comment]
@jfdenise  may you take a look please ?

**Comment by fanste** [comment]
> 
> 
> I think the code should be shared instead of being copied from DeploymentAddHandler

New superclass that is used by `DeploymentAddHandler` and `DeploymentFullReplaceHandler`?

`addFromContentAdditionParameter` is slightly different in both handlers. The only methods, that could be extracted are `loadDeploymentTransformer` and `doTransformation` and of course the variable `deploymentTransformer`.

**Comment by ehsavoie** [comment]
I was thinking of all the transformation code: loading the transformer / transforming in some utility class like DeploymentHandlerUtil

**Comment by jfdenise** [comment]
+1 for the Utility class.

**Comment by fanste** [comment]
For the case, that no notification was sent out for my changes: I've moved the common code to a new utility class.

File: server/src/main/java/org/jboss/as/server/deployment/DeploymentTransformationUtil.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.server.controller.resources.DeploymentAttributes;
+import org.jboss.as.server.deployment.transformation.DeploymentTransformer;
+import org.jboss.as.server.logging.ServerLogger;
+import org.jboss.dmr.ModelNode;
+
+import static org.jboss.as.server.deployment.DeploymentHandlerUtils.getInputStream;
+
+public class DeploymentTransformationUtil {
```
Comment (jfdenise) [inline server/src/main/java/org/jboss/as/server/deployment/DeploymentTransformationUtil.java:35]: Generally xxxUtil class only expose static methods. I would rename it to DeploymentTransformation, instance of this class has a state (the transformer).

File: server/src/main/java/org/jboss/as/server/deployment/DeploymentTransformation.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.server.controller.resources.DeploymentAttributes;
+import org.jboss.as.server.deployment.transformation.DeploymentTransformer;
+import org.jboss.as.server.logging.ServerLogger;
+import org.jboss.dmr.ModelNode;
+
+import static org.jboss.as.server.deployment.DeploymentHandlerUtils.getInputStream;
+
+public class DeploymentTransformation {
```
Comment (bstansberry) [inline server/src/main/java/org/jboss/as/server/deployment/DeploymentTransformation.java:35]: Please make this class package-protected. Thanks :)

**Comment by fanste** [comment]
Rebased the PR and applied the change as requested by @bstansberry

**Review by bstansberry — APPROVED** [review]

**Comment by jmesnil** [comment]
@fanste thanks

**Comment by bstansberry** [comment]
@fanste 

The WildFly project would like to change its license from the GNU Lesser General Public License version 2.1 to the Apache Software License 2.0. We feel that ASL 2.0 is a better option for the project and our user community, as we find a project's use of LGPL can be a barrier to adoption in some parts of the software community, while we see no such barriers from use of ASL.

Do you agree to a relicensing of your contributions to the WildFly, WildFly Core or JBoss AS 7 projects from LGPL 2.1 to ASL 2.0?

I'm happy to answer any questions you may have about this, and I hope you are doing well. Thank you for your contributions to WildFly and open source!

Best regards,
Brian Stansberry
WildFly Project Lead

**Pull request merged**